modules/apps/29-fee/ibc_module_test.go

```diff
@@ -468,3 +470,301 @@ func (suite *FeeTestSuite) TestOnChanCloseConfirm() {
+	testCases := []struct {
+		name          string
+		malleate      func(suite *FeeTestSuite)
+		sourceRelayer bool
+	}{
+		{
+			"success",
+			func(suite *FeeTestSuite) {
+				suite.chainA.GetSimApp().IBCFeeKeeper.SetCounterpartyAddress(suite.chainA.GetContext(), suite.chainA.SenderAccount.GetAddress().String(), suite.chainB.SenderAccount.GetAddress().String())
+			},
+			true,
+		},
+		{
+			"source relayer is empty string",
+			func(suite *FeeTestSuite) {},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+			suite.coordinator.SetupConnections(suite.path)
+
+			// open incentivized channel
+			suite.path.EndpointA.ChanOpenInit()
+			suite.path.EndpointB.ChanOpenTry()
+			suite.path.EndpointA.ChanOpenAck()
+			suite.path.EndpointB.ChanOpenConfirm()
+
+			// set up coins
+			coin := ibctesting.TestCoin
+			fungibleTokenPacket := transfertypes.NewFungibleTokenPacketData(
+				coin.Denom,
+				sdk.NewInt(100).Uint64(),
+				suite.chainA.SenderAccount.GetAddress().String(),
+				suite.chainB.SenderAccount.GetAddress().String(),
+			)
+			packet := channeltypes.NewPacket(
+				fungibleTokenPacket.GetBytes(),
+				suite.chainA.SenderAccount.GetSequence(),
+				suite.path.EndpointA.ChannelConfig.PortID,
+				suite.path.EndpointA.ChannelID,
+				suite.path.EndpointB.ChannelConfig.PortID,
+				suite.path.EndpointB.ChannelID,
+				clienttypes.NewHeight(0, 100),
+				0,
+			)
+
+			// set up & escrow packet fees
+			refundAcc := suite.chainA.SenderAccount.GetAddress()
+			packetId := channeltypes.NewPacketId(suite.path.EndpointA.ChannelID, suite.path.EndpointA.ChannelConfig.PortID, 1)
+			identifiedFee := types.NewIdentifiedPacketFee(
+				packetId,
+				types.Fee{
+					ReceiveFee: validCoins,
+					AckFee:     validCoins2,
+					TimeoutFee: validCoins3,
+				},
+				refundAcc.String(),
+				[]string{},
+			)
+			err := suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), identifiedFee)
+			suite.Require().NoError(err)
+
+			// set up module and callbacks
+			module, _, err := suite.chainA.App.GetIBCKeeper().PortKeeper.LookupModuleByPort(suite.chainA.GetContext(), ibctesting.TransferPort)
+			suite.Require().NoError(err)
+
+			cbs, ok := suite.chainA.App.GetIBCKeeper().Router.GetRoute(module)
+			suite.Require().True(ok)
+
+			// malleate test case
+			tc.malleate(suite)
+
+			result := cbs.OnRecvPacket(suite.chainA.GetContext(), packet, refundAcc)
+
+			if tc.sourceRelayer {
+				suite.Require().NoError(err, "unexpected error for case: %s", tc.name)
+				ack := types.IncentivizedAcknowledgement{
+					Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+					ForwardRelayerAddress: suite.chainB.SenderAccount.GetAddress().String(),
+				}
+				suite.Require().Equal(result, ack)
+			} else {
+				suite.Require().NoError(err, "unexpected error for case: %s", tc.name)
+				ack := types.IncentivizedAcknowledgement{
```

Comment:
if the fee is not enabled, an IncentivizedAcknowledgement should not be returned (this would break backwards compatibility). In the above test case the returned ack should be from ics20

Reply:
this test is just for if the sourceRelayer is an empty string, but I will add one for if the fee is not enabled